### PR TITLE
[OC-748] Remove button log in using keyCloak

### PR DESCRIPTION
**Pull request by freddidierRTE**

### CODE REVIEW

File: ui/main/src/app/app.component.spec.ts [hunk 1]
```diff
@@ -1,178 +0,0 @@
-/* Copyright (c) 2020, RTE (http://www.rte-france.com)
```
Comment (AlexGuironnetRTE) [inline ui/main/src/app/app.component.spec.ts:1]: Why delete all tests for app component (even those unrelated to feature)? Were they too brittle ?

File: ui/main/src/app/services/authentication/authentication.service.ts [hunk 2]
```diff
@@ -96,31 +106,41 @@ export class AuthenticationService {
...
         }
         return new PasswordOrCodeAuthenticationHandler(this, this.store);
     }
 
+
+    regularCheckTokenValidity() {
+        if (this.verifyExpirationDate()) {
+            setTimeout(() => {
+                this.regularCheckTokenValidity();
+            }, 5000);
```
Comment (AlexGuironnetRTE) [inline ui/main/src/app/services/authentication/authentication.service.ts:120]: This means expiration date is checked every 5s ?
If so a named constant might be better.

**Comment by AlexGuironnetRTE** [comment]
# Test 1: CODE FLOW, mode dev -> OK
- Launched dev_environment docker-compose
- Changed access token lifespan to 2 min
- Login
- Existing cards are displayed OK (feed & details)
- Idle wait
- Around 2 min: "You have been disconnected message", click ok, login page
- Log in again, everything looks fine
- Log out ok
- no errors in console/network

**Comment by AlexGuironnetRTE** [comment]
# Test 2: IMPLICIT FLOW, mode dev -> OK

- Deleted and recreated dev_environment docker-compose
- Changed web-ui.yml, re-created dev-env docker and restarted run_all
- Changed access token lifespan for implicit flow to 2 min
- Changed SSO Session Max to 4 min

```
operatorfabric:
  security:
    provider-url: http://localhost:2002
    provider-realm: dev
    base-url: ${operatorfabric.security.provider-url}/auth/realms/${operatorfabric.security.provider-realm}/protocol/openid-connect
    logout-url: ${operatorfabric.security.base-url}/logout?redirect_uri=http://localhost:2002/ui/
    oauth2:
      flow:
        mode: IMPLICIT
        provider: Opfab Keycloak
        delagate-url:  http://localhost:89/auth/realms/${operatorfabric.security.provider-realm}
```

- Login
- Existing cards are displayed OK (feed & details)
- BUT *error trying fetch i18n of 'TEST' version:'1' for locale: 'fr'* (probably unrelated to new dev)
- around 3 min silent refresh request visible in network
- around 4 min 30 silent refresh request visible in network
- 4 min 50 silent refresh in network triggers "you have been disconnected" message
- Click on ok shows login page
- After logging back in, received new cards ok
- Log out ok

**Comment by AlexGuironnetRTE** [comment]
## Test 3: PASSWORD FLOW, mode dev -> KO
- Deleted and recreated dev_environment docker-compose
- Changed web-ui.yml, re-created dev-env docker and restarted run_all
- Changed access token lifespan to 2 min

```
operatorfabric:
  security:
    provider-url: http://localhost:2002
    provider-realm: dev
    base-url: ${operatorfabric.security.provider-url}/auth/realms/${operatorfabric.security.provider-realm}/protocol/openid-connect
    logout-url: ${operatorfabric.security.base-url}/logout?redirect_uri=http://localhost:2002/ui/
    oauth2:
      flow:
        mode: PASSWORD
        provider: Opfab Keycloak
        delagate-url: ${operatorfabric.security.base-url}/auth?response_type=code&client_id=opfab-client
```

No deconnection message, but after a while requests to the back start returning 401 status. -> KO

File: ui/main/src/app/modules/archives/archive.component.spec.ts [hunk 3]
```diff
@@ -46,8 +45,7 @@ describe('ArchivesComponent', () => {
             ],
             providers: [
                 {provide: Store, useClass: Store},
-                {provide: RouterStateSerializer, useClass: CustomRouterStateSerializer},
-                AuthenticationImportHelperForSpecs
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/modules/archives/archive.component.spec.ts:50]: I think it was never needed ( a copy/paste i presume ? )

File: ui/main/src/app/services/authentication/authentication.service.spec.ts [hunk 4]
```diff
@@ -5,502 +5,20 @@
...
     UrlHelperService,
     EventType as OauthEventType,
-    OAuthSuccessEvent
 } from 'angular-oauth2-oidc';
-import createSpyObj = jasmine.createSpyObj;
-import {create} from "domain";
-import any = jasmine.any;
 
 
 export const AuthenticationImportHelperForSpecs = [AuthenticationService,
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/services/authentication/authentication.service.spec.ts:20]: You are right , i will move it global helper file

File: ui/main/src/app/app.component.spec.ts (hunk 1 above)
Comment (freddidierRTE) [reply ui/main/src/app/app.component.spec.ts:1]: To remove button ,i needed de refactor the way authentication is done and refactor the app.component . After that refactor ,unit testing of this component is not needed

**Comment by AlexGuironnetRTE** [comment]
> ## Test 3: PASSWORD FLOW, mode dev -> KO
> 
>     * Deleted and recreated dev_environment docker-compose
> 
>     * Changed web-ui.yml, re-created dev-env docker and restarted run_all
> 
>     * Changed access token lifespan to 2 min
> 
> 
> ```
> operatorfabric:
>   security:
>     provider-url: http://localhost:2002
>     provider-realm: dev
>     base-url: ${operatorfabric.security.provider-url}/auth/realms/${operatorfabric.security.provider-realm}/protocol/openid-connect
>     logout-url: ${operatorfabric.security.base-url}/logout?redirect_uri=http://localhost:2002/ui/
>     oauth2:
>       flow:
>         mode: PASSWORD
>         provider: Opfab Keycloak
>         delagate-url: ${operatorfabric.security.base-url}/auth?response_type=code&client_id=opfab-client
> ```
> 
> No deconnection message, but after a while requests to the back start returning 401 status. -> KO

-> OK, not part of this development, logged OC-890 to do it.

File: ui/main/src/app/services/authentication/authentication.service.ts (hunk 2 above)
Comment (freddidierRTE) [reply ui/main/src/app/services/authentication/authentication.service.ts:120]: Yes , i will add a constant

File: ui/main/src/app/modules/archives/archive.component.spec.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (AlexGuironnetRTE) [reply ui/main/src/app/modules/archives/archive.component.spec.ts:50]: Probably... we've done a lot of it in frontend unit tests unfortunately

**Pull request merged**